packages/leancode_lint/test/lints_test_app/lib/bloc_related_classes_equatable_test.dart

```diff
@@ -0,0 +1,84 @@
+// for tests
+// ignore_for_file: prefer_equatable_mixin, bloc_const_constructors
+
+import 'package:bloc_presentation/bloc_presentation.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+// Bloc-related classes that are not `equatable` are flagged
+class Test1Bloc extends Bloc<Test1Event, Test1State>
+    with BlocPresentationMixin<Test1State, Test1PresentationEvent> {
+  Test1Bloc() : super(Test1State());
+}
+
+final class
+// expect_lint: bloc_related_classes_equatable
+Test1Event {}
+
+final class
+// expect_lint: bloc_related_classes_equatable
+Test1State {}
+
+final class
+// expect_lint: bloc_related_classes_equatable
+Test1PresentationEvent {}
+
+//////////////////////////////////////////////////////////////////////////
+
+// Bloc-related classes that `extend Equatable` are not flagged
+class Test2Bloc extends Bloc<Test2Event, Test2State>
+    with BlocPresentationMixin<Test2State, Test2PresentationEvent> {
+  Test2Bloc() : super(Test2State());
+}
+
+final class Test2Event extends Equatable {
+  @override
+  List<Object?> get props => [];
+}
+
+final class Test2State extends Equatable {
+  @override
+  List<Object?> get props => [];
+}
+
+final class Test2PresentationEvent extends Equatable {
```

Comment:
I am not fully convinced that we should use Equatable if there aren't variables in the state